src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1009,6 +1009,15 @@ object Parsers {
               in.nextToken()
               expr()
             } else EmptyTree
+
+          handler match {
+            case Block(Nil, EmptyTree) => syntaxError(
+              "`catch` block does not contain a valid expression, try adding a case like - `case t: Throwable =>` to the block",
```

Comment:
People should not use `Throwable` unless they know what they're doing, I would suggest `case e: Exception =>` instead